File: src/main/java/antimattermod/core/Energy/IAPMachine.java
package antimattermod.core.Energy;

/**
 * @author C6H2Cl2
 */

public interface IAPMachine {
    int getMaxStoreEnergy();
    boolean canSendEnergy();
    boolean canReceiveEnergy();
    APVoltage getSendVoltage();
    APVoltage getReceiveVoltage();
    int getStoredEnergy();
}

File: src/main/java/antimattermod/core/Energy/IAPMachine.kt
package antimattermod.core.Energy

import net.minecraft.nbt.NBTTagCompound

/**
 * @author C6H2Cl2
 */
open class IAPMachine : IAPReceiver(){
    init {

    }

    override fun writeToNBT(tagCompound: NBTTagCompound) {
        super.writeToNBT(tagCompound)
    }

    override fun readFromNBT(tagCompound: NBTTagCompound) {
        super.readFromNBT(tagCompound)
    }
}